simulator/func_sim/rf/rf.h

```diff
@@ -109,10 +132,12 @@ class RF
         Register reg_num  = instr.get_dst_num();
         bool writes_dst = instr.get_writes_dst();
         auto accumulating_instr = instr.is_accumulating_instr();
+        auto loadlr = instr.is_loadlr();
+        auto offset = instr.get_v_imm();
         if ( !reg_num.is_zero() && writes_dst)
-            write( reg_num, instr.get_v_dst(), accumulating_instr);
+            write( reg_num, instr.get_v_dst(), accumulating_instr, loadlr, offset);
```

Comment:
I advise you to generate mask inside MIPSInstr during LWL/LWR execution and pass it to RF/Memory methods. Current interface is not intuitive at all, i.e. one cannot understand what will happen if offset=2 is passed as option.
Additionally, it will simplify LDL/LDR implementation.